internal/service/ecr/repository.go

```diff
@@ -182,6 +189,12 @@ func resourceRepositoryRead(ctx context.Context, d *schema.ResourceData, meta in
 		d.SetId("")
 		return diags
 	}
+	if err != nil {
+		return create.DiagError(names.ECR, create.ErrActionReading, ResNameRepository, d.Id(), err)
+	}
+	if outputRaw == nil {
+		return create.DiagError(names.ECR, create.ErrActionReading, ResNameRepository, d.Id(), errors.New("empty output"))
+	}
```

Comment:
Shouldn't be necessary as `FindRepository` already does the `nil` check.

Reply:
In theory the added `err` check should prevent this `nil` check from ever executing, but since we do an unchecked type assertion immediately following (which caused the reported panic) it felt reasonable to add for absolute safety.